go/libkb/pgp_key.go

```diff
@@ -607,6 +609,13 @@ func (k *PGPKeyBundle) CheckFingerprint(fp *PGPFingerprint) error {
 }
 
 func (k *PGPKeyBundle) SignToString(msg []byte) (sig string, id keybase1.SigID, err error) {
+	if k.Entity.PrivateKey == nil || k.Entity.PrivateKey.PrivateKey == nil {
```

Comment:
help.go

On Thu, Mar 24, 2016 at 2:36 PM, Jacob H. Haven <notifications@github.com>
wrote:

> In go/libkb/pgp_key.go
> <https://github.com/keybase/client/pull/2440#discussion_r57366433>:
>
> > @@ -607,6 +609,13 @@ func (k *PGPKeyBundle) CheckFingerprint(fp *PGPFingerprint) error {
> >  }
> >
> >  func (k *PGPKeyBundle) SignToString(msg []byte) (sig string, id keybase1.SigID, err error) {
> > +	if k.Entity.PrivateKey == nil || k.Entity.PrivateKey.PrivateKey == nil {
>
>
>    - This should only happen when we've imported a PGP key with dummy
>    skeys (i.e. from GPG)
>    - I've added a commit that bails out of any GPG CLI call if the path
>    hasn't been properly set by GPGCLI#Configure. It will fail cleanly,
>    there's nothing better we can do.
>    - Again, this is only for a key with specific GPG dummy key
>    extensions. It *will fail* if the smart card isn't connected/we're on
>    a different machine without that GPG key, but there's no real way for us to
>    know without calling out to GPG.
>
> @patrickxb <https://github.com/patrickxb> the gpg-usage docs probably do
> need to be updated. Where can I find those?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/2440/files/4da64d43f0c07b2309fe089ffc59eb6fa373c0e9#r57366433>
>